File: prototype/rails/src/test/delivery/flash_magic_link_delivery_test.rb
require "test_helper"

class FlashMagicLinkDeliveryTest < ActiveSupport::TestCase
  test "it hands the link to the debug alert the layouts render" do
    flash = ActionDispatch::Flash::FlashHash.new

    FlashMagicLinkDelivery.new(flash).deliver(email: "artist@example.com", url: "http://localhost:3000/auth/magic/abc")

    assert_equal "http://localhost:3000/auth/magic/abc", flash[:debug_magic_link]
  end
end

File: prototype/node/src/app/delivery/flash-magic-link-delivery.test.ts
import { test } from 'node:test'
import assert from 'node:assert/strict'
import { flashMagicLinkDelivery } from './flash-magic-link-delivery.ts'
import { openCommerceWorld } from '../test/commerce-world.ts'

const LINK = {
  email: 'artist@example.com',
  url: 'http://localhost:4000/auth/magic/abc',
  actorType: 'seller',
} as const

test('it hands the link to the debug alert the layouts render', async (t) => {
  const world = await openCommerceWorld()
  t.after(world.close)

  const flash = await flashMagicLinkDelivery.deliver(world.context, LINK)

  assert.deepEqual(flash, { debugMagicLink: 'http://localhost:4000/auth/magic/abc' })
})

test('it queues nothing: the link never leaves the request', async (t) => {
  const world = await openCommerceWorld()
  t.after(world.close)

  await flashMagicLinkDelivery.deliver(world.context, LINK)

  const queued = await world.db.selectFrom('outboxMessages').selectAll().execute()
  assert.equal(queued.length, 0)
})
